src/main/kotlin/com/evalify/evalifybackend/batch/controller/BatchController.kt

```
@@ -20,4 +35,161 @@ class BatchController(val batchBankService: BatchBankService) {
+    @GetMapping("/search")
+    fun searchBatches(
+        @RequestParam query: String,
+        @RequestParam(required = false, defaultValue = "0") page: Int,
+        @RequestParam(required = false, defaultValue = "10") size: Int,
+        @RequestParam(required = false) sort_by: String?,
+        @RequestParam(required = false) sort_order: String?
+    ): ResponseEntity<PaginatedResponse<BatchResponse>> {
+        return try {
+            val batches = batchService.searchBatches(query, page, size, sort_by, sort_order)
+            ResponseEntity.ok(batches)
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)